src/plugins/intel_cpu/src/emitters/snippets/brgemm_base.hpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2020-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "emitters/snippets/cpu_kernel_executor_table.hpp"
+#include "emitters/utils.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "snippets/lowered/loop_info.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "utils/general_utils.h"
+
+namespace ov::intel_cpu {
+
+struct BrgemmGenericKernelConfig : public snippets::KernelExecutorBase::GenericConfig {
+public:
+    BrgemmGenericKernelConfig() = default;
+
+    bool is_completed() const override;
+    bool is_empty() const;
+
+    void update(int64_t M, int64_t N, int64_t K, int64_t LDA, int64_t LDB, int64_t LDC, float beta);
```

Comment:
Do you think it's worth to make this method virtual? We could override it in 2 places already

Reply:
Yes, I think `virtual `is better, more easy to understand by avoiding confusion and remove redundancy, done! Thank you Vladislav!